onadata/apps/restservice/signals.py

```diff
@@ -24,7 +25,14 @@ def call_webhooks(sender, **kwargs):  # pylint: disable=unused-argument
     if ASYNC_POST_SUBMISSION_PROCESSING_ENABLED:
         call_service_async.apply_async(args=[instance_id], countdown=1)
     else:
-        call_service(instance_id)
+        try:
+            instance = Instance.objects.get(pk=instance_id)
+        except Instance.DoesNotExist:
+            # if the instance has already been removed we do not send it to the
+            # service
```

Comment:
We should probably check the master database instead. I don't expect the instance to be deleted before the request is fully processed; _signals actually need to be completed before a response is returned_